### PR TITLE
Add ServerSuppressionsChangedHandler

### PR DESCRIPTION
**Pull request by bigfluffycookie**
fixes https://github.com/SonarSource/sonarlint-visualstudio/issues/3819

### CODE REVIEW

File: src/ConnectedMode/Suppressions/ServerSuppressionsChangedHandler.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        private void ServerSuppressionsChanged(object sender, EventArgs e)
+        {
+            clientSuppressionSynchronizer.SynchronizeSuppressedIssues();
+        }
+
+        public void Dispose()
+        {
+            suppressedIssuesMonitor.SuppressionsUpdateRequested -= ServerSuppressionsChanged;
+        }
+
```
Comment (bigfluffycookie) [inline src/ConnectedMode/Suppressions/ServerSuppressionsChangedHandler.cs:52]: Missed this, will fix it with other pr errors

File: src/ConnectedMode/Suppressions/ServerSuppressionsChangedHandler.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            this.clientSuppressionSynchronizer = clientSuppressionSynchronizer;
+            this.suppressedIssuesMonitor = suppressedIssuesMonitor;
+
+            suppressedIssuesMonitor.SuppressionsUpdateRequested += ServerSuppressionsChanged;
+        }
+
+        private void ServerSuppressionsChanged(object sender, EventArgs e)
+        {
+            clientSuppressionSynchronizer.SynchronizeSuppressedIssues();
```
Comment (bigfluffycookie) [inline src/ConnectedMode/Suppressions/ServerSuppressionsChangedHandler.cs:45]: Note here to maybe add logging

**Comment by duncanp-sonar** [comment]
@bigfluffycookie, just to check - this PR is just about the last commit?

**Comment by bigfluffycookie** [comment]
@duncanp-sonar 
Yes, the other two commits are from the automatic branch change.

File: src/ConnectedMode/Suppressions/ServerSuppressionsChangedHandler.cs (hunk 2 above)
Comment (duncanp-sonar) [reply src/ConnectedMode/Suppressions/ServerSuppressionsChangedHandler.cs:45]: Either here, or in the synchronizer (or perhaps both).
We need to decide what information would be useful in normal operation (e.g. "_Re-apply server-side suppressions_") and what would help in troubleshooting/support cases (e.g. "x issues suppressed/updated").
I'll add a task to the board.

**Pull request merged**